src/documentation/io/authoring-overview.md

```diff
@@ -10,35 +10,76 @@ permalink: /documentation/io/authoring-overview/
+1. Splitting the data into parts to be read in parallel
+2. Reading from each of those parts.
+
+Each of those steps will be a ParDo, with a GroupByKey in between. The GroupByKey is an implementation detail but for most runners it allows the runner to use different numbers of workers for determing how to split up the data to be read into chunks (which will likely occur on very few workers) and reading (which will likely benefit from more workers.) For runners which support Dynamic Work Rebalancing, the GBK will also allow that to occur.
+
+Here are some examples of read transform implementations using the "reading as a mini-pipeline" model to use when data can be read in parallel:
+* **Reading from chunks of a file** - if you have a file format that can be sub-divided, the structure will look like:
+  * Determine Byte Ranges ParDo: As input, takes in a file name. Emit a PCollection of byte ranges inside of the file.
+  * Read Byte Range ParDo: Given the PCollection of byte ranges, and reads each byte range and reads the data in that byte range, emitting a PCollection of records.
+* **Reading from a file glob** - For example reading all files in "~/data/**"
+  * Get File Paths ParDo: As input, take in a file glob. Emit a PCollection of strings, each of which is a file path.
+  * Reading ParDo: Given the PCollection of file paths, read each one, emitting a PCollection of records.
+* **Reading from a NoSQL Database** (eg Apache HBase) - these databases often allow reading from ranges in parallel.
+  * Determine Key Ranges ParDo: As input, receive connection information for the database and the key range to read from. Output a PCollection of key ranges that can be read in parallel efficiently.
+  * Read Key Range ParDo: Given the PCollection of key ranges, read the key range, emitting a PCollection of records.
+
+For data stores or files where reading cannot occur in parallel, reading is a simple task that can be accomplished with a single ParDo. For example:
+* **Reading from a database query** - traditional SQL database queries often can only be read in sequence. The ParDo in this case would establish a connection to the database and read batches of records, emitting a PCollection of those records.
+* **Reading from a gzip file** - a gzip file has to be read in order, so it cannot be parallelized. The ParDo in this case would open the file and read in sequence, emitting a PCollection of records from the file.
+
 
 ### When to implement using the Source API
-TODO
+The above discussion is in terms of ParDos - this is because Sources have proven to be tricky to implement. At this point in time, **it's not recommended that you implement a Source if you can get away with a ParDo**. If you're trying to decide between the two, feel free to email the Beam dev mailing list and we can discuss the specific pros and cons of your case.
```

Comment:
Clarify that deriving from FileBasedSource is still fine, if you're reading from byte ranges.
Also, quote the address of the dev mailing list.